player/lua.c

```diff
@@ -879,7 +872,7 @@ static void pushnode(lua_State *L, mpv_node *node)
         lua_pushstring(L, node->u.string);
         break;
     case MPV_FORMAT_INT64:
-        lua_pushnumber(L, node->u.int64);
+        lua_pushinteger(L, node->u.int64);
```

Comment:
lua integer is defined as `ptrdiff_t` so it would be 32-bit integer on 32-bit builds. I think it is more correct to keep using `lua_pushnumber` (`double`) to support `int64` input that we have here.

Reply:
Yeah I suspected as much.